hw9/modules/player.ts

```diff
@@ -0,0 +1,90 @@
+import { Library } from "./library";
+import { Track } from "./track";
+import {EventEmitter} from 'events';
+export class Player {
+    public TracksPlayer: Array<Track>= [];
+    private nextTrack: Function;
+    private prevTrack: Function;
+    private pauseTrack: Function;
+    private static instance: Player;
+    private constructor(library: Library) {};
+    static getInstance(library: Library): Player {
+        if (!Player.instance) {
+            Player.instance = new Player(library);
+            library.tracksList().forEach(element => {
+                Player.instance.TracksPlayer.push(element);
+            });                                        
+        }
+        return Player.instance;
+    };
+    //PLAY METHOD
+    play():string{    
+        const player: EventEmitter = new EventEmitter();                  
+        let current: number = 0;         
+        let strInfo: string = `${Player.instance.TracksPlayer[current].name}, Duration: ${Player.instance.TracksPlayer[current].durationTime}`; 
+        //EventEmitter next()        
+        this.nextTrack = function():void{player.once('next', ()=>{
+            if(current + 1 >= Player.instance.TracksPlayer.length){ 
+                current = 0;                
+            }
+            else{
+                current++;                            
+            }   
+                    
+        })};
+        //EventEmitter prev()    
+        this.prevTrack = function():void{player.once('prev', ()=>{
```

Comment:
commands like prev, next and pause - is worked only once

Reply:
fixed